Projects/19-Captsone_01/ispy1/predictive_statistics.py

```diff
@@ -0,0 +1,255 @@
+from sklearn import preprocessing
+from sklearn import metrics
+from sklearn.model_selection import train_test_split, GridSearchCV
+from sklearn import metrics, linear_model
+from imblearn import over_sampling
+from sklearn.calibration import CalibratedClassifierCV
+from sklearn.ensemble import RandomForestClassifier as RFC
+import numpy as np
+import matplotlib.pyplot as plt
+
+RANDOM_STATE = 42; # for reproducibility
+
+
+def labels_to_numbers(DataFrame, Variable):
+    le = preprocessing.LabelEncoder()
+    numbers_ = le.fit_transform(DataFrame[Variable].values)
+    return numbers_
+
+
+def binary_classifier_metrics(classifier, Xtrain, Ytrain, Xtest, Ytest):
+    # metrics
+    predicted_class = classifier.predict(Xtest)
+
+    kappa = metrics.cohen_kappa_score(Ytest,predicted_class)
+    np.round(kappa,3)
+
+    auc = metrics.roc_auc_score(Ytest,predicted_class)
+    auc = np.round(auc,3)
+
+
+    # ROC curve
+    probability = classifier.predict_proba(Xtest)
+    fpr, tpr, _  = metrics.roc_curve(Ytest, probability[:,1], drop_intermediate=False)
+
+    # report
+    print(metrics.classification_report(Ytest,predicted_class))
+    print('The estimated Cohen kappa is ' + str(kappa))
+    print('The estimated AUC is ' + str(auc))
+    print('=='*30)
+    print('\n'*2)
+
+    return auc, kappa, fpr, tpr
+
+
+def split_data(Xdata,Ydata, oversample, K_neighbors = 4):
+    if oversample == False:
+        X_train, X_test, y_train, y_test = train_test_split(Xdata, Ydata,
+                                                            train_size = 0.70,
+                                                            random_state=RANDOM_STATE)
+    elif oversample == True:
+        print('Data was oversampled using the ADASYN method')
+        smote = over_sampling.ADASYN(random_state = RANDOM_STATE, n_neighbors = K_neighbors)
+        # split
+        X_train, X_test, y_train, y_test = train_test_split(Xdata, Ydata,train_size = 0.70, random_state=RANDOM_STATE)
```

Comment:
I'd make `train_size` a parameter like `RANDOM_STATE` and pass it to the function.

more general example:
```
Y = 5

def f(x, y=Y):
    x + y
```

This way your function `f` depends on `x` and `y` so you can test it regardless of the value of `Y`. Removing this dependency on `Y` makes your code easier to test, reason about, and maintain.